enumerable.rb

```diff
@@ -0,0 +1,149 @@
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i])
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i], i)
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
```

Comment:
- [ ] calls the given block once for each element in self
- [ ] returns an enumerator if no block is given